File: contract-service/src/domain/ContractStatus.ts
/**
 * Value Object: ContractStatus
 * สถานะของสัญญา ไม่มี identity ของตัวเอง กำหนดครั้งเดียว
 */
export const ContractStatus = {
  DRAFT: 'DRAFT',
  PENDING_SIGN: 'PENDING_SIGN',
  SIGNED: 'SIGNED',
  CANCELLED: 'CANCELLED',
} as const;

export type ContractStatus = (typeof ContractStatus)[keyof typeof ContractStatus];

File: _java-archive/kafka-listener/src/main/java/dev/kritchalach/kafka/domain/ContractStatus.java
package dev.kritchalach.kafka.domain;

/**
 * Value Object: ContractStatus
 * สถานะของสัญญา ไม่มี identity ของตัวเอง กำหนดครั้งเดียว
 */
public enum ContractStatus {
    DRAFT,
    PENDING_SIGN,
    SIGNED,
    CANCELLED
}
